lib/ui/pages/news_list/news_list_page.dart

```diff
@@ -27,46 +27,60 @@ class _NewsPageState extends State<NewsListPage> {
         child: Column(
           crossAxisAlignment: CrossAxisAlignment.start,
           children: [
-            PrimaryNewsListItem(
-              news: PrimaryNewsListEntity(
-                imageUrl: 'https://i.picsum.photos/id/9/800/800.jpg',
-                title: "New laptops seem to be stuck to the board",
-                authorName: "Marek Aureliusz",
-                authorImageUrl:
-                    'https://www.ludoviccareme.com/files/image_211_image_fr.jpg',
-              ),
+            Container(
+              width: double.maxFinite,
+              height: 80,
+            ),
+            _buildNewsSectionHeader(
+              headerTitle: Strings.mostViewed,
+              routeId: null,
+            ),
+            SizedBox(height: _Constants.sectionHeaderPadding),
+            PrimaryNewsListView(
+              primaryNewsListEntities: primaryNewsListEntities.take(5).toList(),
+            ),
+            _buildNewsSectionHeader(
+              headerTitle: Strings.newest,
+              routeId: SecondaryNewsListPage.id,
             ),
-            _buildSecondaryNewsSectionHeader(),
             SizedBox(height: _Constants.sectionHeaderPadding),
             SecondaryNewsListView(
-                secondaryNewsListEntities: secondaryNewsListEntities.take(3).toList(),
+              secondaryNewsListEntities: secondaryNewsListEntities.take(3).toList(),
             ),
           ],
         ),
       ),
     );
   }
 
-  Widget _buildSecondaryNewsSectionHeader() => Padding(
-        padding: EdgeInsets.symmetric(horizontal: Dimens.pagePadding),
-        child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          crossAxisAlignment: CrossAxisAlignment.center,
-          children: [
-            Hero(
-              tag: Strings.newest,
-              child: Text(
-                Strings.newest,
-                style: Theme.of(context).textTheme.headline3,
-              ),
-            ),
-            PrimaryTextButton(
-              text: Strings.showAll,
-              onPressed: () {
-                Navigator.pushNamed(context, SecondaryNewsListPage.id);
-              },
+  Widget _buildNewsSectionHeader({
+    @required String headerTitle,
+    @required String routeId,
+  }) {
+    return Padding(
+      padding: EdgeInsets.symmetric(horizontal: Dimens.pagePadding),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          Hero(
```

Comment:
Klikając na See all dla pionowej listy wywala się 'There are multiple heroes that share the same tag within a subtree'

Reply:
to dlatego, że dałem nulla w roucie, który ma przechodzić do kolejnego ekranu. Wynika to z tego, że ten ekran to już osobny task, a chciałem reużyć widgetu stworzonego przez Ciebie. Wziąłem już taska na stworzenie tego ekranu i tam to fixnę, ok? Ale tak, powinienem to oznaczyć jakimś //FIXME

Reply:
Ach spoko, kompletnie się nie zagłębiałem w to z czego to wynika i nie pomyślałem że zamierzone, myślałem, że może jakieś mocki mają identyczny tytuł czy coś :P